### DISCUSSION TITLE
Snapclient not working

### DISCUSSION THREAD

**Discussion by skycamel** [rern/rAudio#1094] [General]
I'm truing to get snapclient on my second rpi to work but there seem to be some error.
On the server side I get following error

snapserver.service- Snapcast server
     Loaded: loaded (/usr/lib/systemd/system/snapserver.service; enabled; preset: disabled)
     Active: active (running) since Wed 2022-11-16 13:35:36 CET; 1 day 4h ago
       Docs: man:snapserver(1)
   Main PID: 309 (snapserver)
      Tasks: 5 (limit: 1062)
        CPU: 1h 1min 41.799s
     CGroup: /system.slice/snapserver.service
             └─309 /usr/bin/snapserver --logging.sink=system --server.datadir=/var/lib/snapserver

Nov 17 17:47:57 rAudio snapserver[309]: StreamServer::NewConnection: 192.168.88.175
Nov 17 17:47:57 rAudio snapserver[309]: StreamServer::NewConnection: 192.168.88.175
Nov 17 17:47:57 rAudio snapserver[309]: Hello from b8:27:eb:b3:b7:c7, host: rAudio-Amp, v0.26.0, ClientName: Snapclient, OS: Arch Linux ARM, Arch: aarch64, Protocol version: 2
Nov 17 17:47:57 rAudio snapserver[309]: Error reading message header of length 0: End of file
Nov 17 17:47:57 rAudio snapserver[309]: onDisconnect: b8:27:eb:b3:b7:c7
Nov 17 17:47:57 rAudio snapserver[309]: Hello from b8:27:eb:b3:b7:c7, host: rAudio-Amp, v0.26.0, ClientName: Snapclient, OS: Arch Linux ARM, Arch: aarch64, Protocol version: 2
Nov 17 17:47:57 rAudio snapserver[309]: StreamSession write error (msg length: 0): Broken pipe
Nov 17 17:47:57 rAudio snapserver[309]: Error in socket shutdown: Transport endpoint is not connected
Nov 17 17:48:20 rAudio snapserver[309]: State changed: default, state: idle => playing
Nov 17 17:48:20 rAudio snapserver[309]: onStateChanged (default): playing

Can anyone help me please?
Thanks

**Comment by cmh714** [comment]
I got this running a pacman update, might be related?? regardless, it may become an issue so @rern just an FYI (I dont use it so no problem for me)

(1/1) upgrading snapcast                           [######################] 100%
:: The default setup will create a pipe /tmp/snapfifo.
   Due to recent changes in systemd, pipes in /tmp are by default only
   writable by the owning user (here: sysuser snapserver).
   This is controlled by systclt fs.protected_fifos.
   A possible workaround is to disable the sysctl feature by running:
     # sysctl fs.protected_fifos=0
:: Snapcast now has a built-in snapweb control client which is enabled
   by default on a new setup.  This functionality enables a webserver
   on port 1780.  This can be controlled by the doc_root variable in
   /etc/snapserver.conf.

**Reply by rern** [reply]
`pacman -Syu` upgraded `flac` to a new version. `sanpcast` has been just upgraded to fix the bug. 
Try `pacman -Syu` again.

**Reply by skycamel** [reply]
Did the upgrade, unfortunately still the same problem. I only upgraded the snapcastclient rpi side, not the server (not enough space on SD)
But my other snapcast clients (linux, AFTV) connect without problem. (see log cubieboard connected)
Any suggestions,
Luc

Nov 18 13:42:02 rAudio snapserver[333]: StreamServer::NewConnection: 192.168.88.75
Nov 18 13:42:02 rAudio snapserver[333]: Hello from 02:50:01:c2:8d:51, host: cubieboard, v0.24.0, ClientName: Snapclient, OS: Debian GNU/Linux 10 (buster), Arch: armv7l, Protocol version: 2
Nov 18 17:19:01 rAudio snapserver[333]: StreamServer::NewConnection: 192.168.88.175
Nov 18 17:19:01 rAudio snapserver[333]: StreamServer::NewConnection: 192.168.88.175
Nov 18 17:19:01 rAudio snapserver[333]: Hello from b8:27:eb:b3:b7:c7, host: rAudio-Amp, v0.26.0, ClientName: Snapclient, OS: Arch Linux ARM, Arch: aarch64, Protocol version: 2
Nov 18 17:19:01 rAudio snapserver[333]: Error reading message header of length 0: End of file
Nov 18 17:19:01 rAudio snapserver[333]: onDisconnect: b8:27:eb:b3:b7:c7
Nov 18 17:19:01 rAudio snapserver[333]: Hello from b8:27:eb:b3:b7:c7, host: rAudio-Amp, v0.26.0, ClientName: Snapclient, OS: Arch Linux ARM, Arch: aarch64, Protocol version: 2
Nov 18 17:19:01 rAudio snapserver[333]: StreamSession write error (msg length: 0): Broken pipe
Nov 18 17:19:01 rAudio snapserver[333]: Error in socket shutdown: Transport endpoint is not connected


Here you have the log on the client side
snapclient.service - Snapcast client
     Loaded: loaded (/usr/lib/systemd/system/snapclient.service; disabled; preset: disabled)
     Active: active (running) since Fri 2022-11-18 17:19:00 CET; 4min 3s ago
       Docs: man:snapclient(1)
   Main PID: 1712 (snapclient)
      Tasks: 2 (limit: 1062)
        CPU: 7.587s
     CGroup: /system.slice/snapclient.service
             └─1712 /usr/bin/snapclient --logsink=system --latency=100

Nov 18 17:19:01 rAudio-Amp snapclient[1712]: Codec: flac, sampleformat: 48000:16:2
Nov 18 17:19:01 rAudio-Amp snapclient[1712]: Player name: alsa, device: default:CARD=Headphones, description: bcm2835 Headphones, bcm2835 Headphones
Nov 18 17:19:01 rAudio-Amp snapclient[1712]: Default Audio Device, idx: 1, sharing mode: unspecified, parameters: 
Nov 18 17:19:01 rAudio-Amp snapclient[1712]: Mixer mode: software, parameters: 
Nov 18 17:19:01 rAudio-Amp snapclient[1712]: Sampleformat: 48000:16:2, stream: 48000:16:2
Nov 18 17:19:01 rAudio-Amp snapclient[1712]: Using default buffer_time: 80 ms, default fragments: 4
Nov 18 17:19:01 rAudio-Amp snapclient[1712]: PCM name: default:CARD=Headphones, sample rate: 48000 Hz, channels: 2, buffer time: 80000 us, periods: 4, period time: 20000 us, period frames: 960
Nov 18 17:19:01 rAudio-Amp snapclient[1712]: No chunks available
Nov 18 17:19:01 rAudio-Amp snapclient[1712]: Failed to get chunk
Nov 18 17:19:01 rAudio-Amp snapclient[1712]: diff to server [ms]: 1.84739e+07
